File: leet_code/two_sum.rb
def two_sum(nums, target)
  counter_hash = Hash.new(0)
  index_hash = Hash.new { |h, k| h[k] = [] }
  nums.each_with_index do |num, i|
    counter_hash[num] += 1
    index_hash[num] << i
    if num * 2 == target
      return index_hash[num] if counter_hash[num] == 2
    elsif counter_hash[target - num] > 0
      return [index_hash[num][0], index_hash[target - num][0]]
    end
  end
end

File: leet_code/go/twoSum.go
func twoSum(nums []int, target int) []int {
	numHash := make(map[int]int)
	for i := 0; i < len(nums); i++ {
		complement := target - nums[i]
		if val, ok := numHash[complement]; ok {
			indices := []int{val, i}
			return indices
		} else {
			numHash[nums[i]] = i
		}
	}
	return []int{0, 0}
}
